categorize_notify/program.py

```diff
@@ -10,60 +10,86 @@
 
 import base64
 import time
+from datetime import datetime, timezone
 
 import autokitteh
 from autokitteh import google, openai, slack
 
 
 SLACK_CHANNELS = ["demos", "engineering", "ui"]
 
+processed_message_ids = set()
+start_time = datetime.now(timezone.utc)
+
 
 def on_http_get(event):
-    total_messages = None
     while True:
-        total_messages = _poll_inbox(total_messages)
+        _poll_inbox()
         time.sleep(10)
 
 
 @autokitteh.activity
-def _poll_inbox(prev_total: int):
+def _poll_inbox():
     gmail = google.gmail_client("my_gmail").users()
-    curr_total = gmail.getProfile(userId="me").execute()["messagesTotal"]
-    # Note: This is not meant to be a robust solution for handling email operations.
-    if prev_total and curr_total > prev_total:
-        new_email_count = curr_total - prev_total
-        message_ids = _get_latest_message_ids(gmail, new_email_count)
-        for message_id in message_ids:
-            _process_email(gmail, message_id)
+    token = None
+    current_message_ids = set()
+
+    while True:
+        query = "in:inbox -in:drafts"
+        if token:
```

Comment:
my guess is that only the inside of the loop needs to be an activity, not the entire loop